### ISSUE TITLE
Concretize output improvement

### ISSUE THREAD

**Issue by jedwards4b** [LLNL/spack#35005]
### Summary

It would be nice if spack clearly identified in the concretize step packages which are 
external and will not be built vs packages that will be built from source. 
In this case I was trying to use mvapich2 and had defined it as an external
but spack insisted on building from source.   After several attempts at changing the spec and packages I was able to get it
to use the installed library, but I couldn't determine that in the concretize step - only once spack was in the install step could I determine whether the installed mvapich2 library would be used. 

### Rationale

I've added the mvapich2 library as an external in my .spack/packages.yaml
```
/home/jedwards/.spack/packages.yaml:12                      mvapich2:
/home/jedwards/.spack/packages.yaml:13                        externals:
/home/jedwards/.spack/packages.yaml:14                        - spec: mvapich2@2.3.3
/home/jedwards/.spack/packages.yaml:15                          module: [mpi/2.3.3/intel/20.0.1]
/home/jedwards/.spack/packages.yaml:16                        buildable: False
```
yet when I concretize spack doesn't find it and wants to build from source:
```
==> Concretized hdf5@1.12.2%intel ^mvapich2@2.3.3%intel
 -   kesrheo  hdf5@1.12.2%intel@20.0.1~cxx~fortran~hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=RelWithDebInfo arch=linux-centos7-skylake_avx512
[+]  o45orep      ^cmake@3.19.0-rc2%intel@20.0.1~doc+ncurses+ownlibs~qt build_system=generic build_type=Release patches=b48396c arch=linux-centos7-skylake_avx512
 -   drge4rt      ^mvapich2@2.3.3%intel@20.0.1~alloca~cuda~debug+regcache+wrapperrpath build_system=autotools ch3_rank_bits=32 fabrics=mrail file_systems=auto process_managers=auto threads=multiple arch=linux-centos7-skylake_avx512
[+]  opvt3nb      ^pkgconf@1.8.0%intel@20.0.1 build_system=autotools arch=linux-centos7-skylake_avx512
[+]  vxhurhg      ^zlib@1.2.13%intel@20.0.1+optimize+pic+shared build_system=makefile arch=linux-centos7-skylake_avx512
```

### Description

Discussed on [slack](https://spackpm.slack.com/archives/C5W7NKZJT/p1674054945895519)

### Additional information

spack --version
0.20.0.dev0 (2fdd72094aac5346684a80b5f18285a06b218eb8)

### General information

- [X] I have run `spack --version` and reported the version of Spack
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by alalazo** [comment]
There is a misunderstanding here, even though I can see why it may cause confusion:
1. The fact that `mvapich2` appears without dependencies, means that the external will be used. 
2. The `[+]` symbol means instead that the spec appears in the DB already. 

This is somehow due to how externals are treated right now, i.e. as "abstract nodes" that are concretized and whose dependencies are trimmed. There is a lot of heuristic in treating externals like that, and maybe we can be clearer in our `spec` output.

**Comment by jedwards4b** [comment]
I think that I understand how to read this now, thank you.

**Comment by jedwards4b** [comment]
Closing as duplicate of  https://github.com/spack/spack/pull/33792
